### PR TITLE
feat: Add deprecated HPA (autoscaling/v2beta1/2)

### PR DESCRIPTION
**Pull request by stgrace**

### CODE REVIEW

File: pkg/rules/rego/deprecated-1-25.rego [hunk 1]
```diff
@@ -46,6 +46,11 @@ deprecated_api(kind, api_version) = api {
 			"new": "batch/v1",
 			"since": "1.21",
 		},
+		"HorizontalPodAutoscaler": {
+			"old": ["autoscaling/v2beta1"],
+			"new": "autoscaling/v2",
+			"since": "1.23",
```
[the remark this answers is not in the document]
Comment (stepanstipl) [reply pkg/rules/rego/deprecated-1-25.rego:52]: Also, looks like there was `autoscaling/v2beta2`, might be worth adding it too + a test.

**Review by stepanstipl — CHANGES_REQUESTED** [review]
Many thanks @stgrace 👍 . In general, it looks good to me, just the note about code org. as this looks like it belongs in `1.26`. Other than that, the CI currently fails on some minor formatting issues, please see [^1], [^2].

Also, as to commit messages, I would suggest using the `feat: ` prefix, as this PR adds new functionality.

_Note: dupe of #330, related to #356_

[^1]: https://github.com/doitintl/kube-no-trouble/actions/runs/3130260911/jobs/5107405014
[^2]: https://github.com/doitintl/kube-no-trouble/actions/runs/3130260911/jobs/5107405540

File: pkg/rules/rego/deprecated-1-25.rego (hunk 1 above)
[the remark this answers is not in the document]
Comment (dark0dave) [reply pkg/rules/rego/deprecated-1-25.rego:52]: I can add
its depricated in 25 too?

[the remark this answers is not in the document]
Comment (stgrace) [reply pkg/rules/rego/deprecated-1-25.rego:52]: Alright,
So according to the docs it is actually deprecated in 1.25 (atleast autoscaling/v2beta1 is)
I can add autoscaling/v2beta2 to a new deprecated file already :)

[the remark this answers is not in the document]
Comment (stepanstipl) [reply pkg/rules/rego/deprecated-1-25.rego:52]: > So according to the docs it is actually deprecated in 1.25 (atleast autoscaling/v2beta1 is)
> I can add autoscaling/v2beta2 to a new deprecated file already :)

Ah, sorry, I missed that, thanks for pointing it out. Currently, it wouldn't work to have the same resource twice in two separate files (we're thinking about improving this). You can have multiple deprecated versions (`old`) within one resource definition.

**Comment by stepanstipl** [comment]
Thanks a lot, @stgrace 👍 

I think both `v2beta1` and `v2beta2` autoscaling APIs have bow been covered via https://github.com/doitintl/kube-no-trouble/pull/368, please correct me if that's not correct or missing anything there.

_(We had like 3 different PRs for this, should have all been closed when #368 was merged, I'm closing them now)_

**Pull request closed without merging**